src/plugins/discover/common/utils/sorting/get_es_query_sort.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { EsQuerySortValue, SortDirection } from '@kbn/data-plugin/public';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import type { IUiSettingsClient } from '@kbn/core-ui-settings-browser';
+import { CONTEXT_TIE_BREAKER_FIELDS_SETTING } from '@kbn/discover-utils';
+
+/**
+ * Returns `EsQuerySort` which is used to sort records in the ES query
+ * https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-sort.html
+ * @param sortDir
+ * @param timeFieldName
+ * @param tieBreakerFieldName
+ * @param isTimeNanosBased
+ */
+export function getEsQuerySort({
+  sortDir,
+  timeFieldName,
+  tieBreakerFieldName,
+  isTimeNanosBased,
+}: {
+  sortDir: SortDirection;
+  timeFieldName: string;
+  tieBreakerFieldName: string;
+  isTimeNanosBased: boolean;
+}): [EsQuerySortValue, EsQuerySortValue] {
+  return [
+    getESQuerySortForTimeField({ sortDir, timeFieldName, isTimeNanosBased }),
+    getESQuerySortForTieBreaker({ sortDir, tieBreakerFieldName }),
+  ];
+}
+
+/**
+ * Prepares "sort" structure for a time field for next ES request
+ * @param sortDir
+ * @param timeFieldName
+ * @param isTimeNanosBased
+ */
+export function getESQuerySortForTimeField({
+  sortDir,
+  timeFieldName,
+  isTimeNanosBased,
+}: {
+  sortDir: SortDirection;
+  timeFieldName: string;
+  isTimeNanosBased: boolean;
+}): EsQuerySortValue {
+  return {
+    [timeFieldName]: {
+      order: sortDir,
+      ...(isTimeNanosBased
+        ? {
+            format: 'strict_date_optional_time_nanos',
+            numeric_type: 'date_nanos',
+          }
+        : { format: 'strict_date_optional_time' }),
+    },
+  };
+}
+
+/**
+ * Prepares "sort" structure for a tie breaker for next ES request
+ * @param sortDir
+ * @param tieBreakerFieldName
+ */
+export function getESQuerySortForTieBreaker({
+  sortDir,
+  tieBreakerFieldName,
+}: {
+  sortDir: SortDirection;
+  tieBreakerFieldName: string;
+}): EsQuerySortValue {
+  return { [tieBreakerFieldName]: sortDir };
+}
+
+/**
+ * The list of field names that are allowed for sorting, but not included in
+ * data view fields.
+ */
+const META_FIELD_NAMES: string[] = ['_seq_no', '_doc', '_uid'];
```

Comment:
@weltenwort Thanks for bring our attention to this! I have not introduced this array, just moved from context utils to common utils. For supporting legacy behaviour, I think we should keep `_uid` available for user's selection. We can also add `_shard_doc` to the list, wdyt?

Reply:
I think we can only use `_shard_doc` within a PIT context.

Reply:
```
The _shard_doc value is the combination of the shard index within the PIT and the Lucene’s internal doc ID, it is unique per document and constant within a PIT.
```
wouldn't `_doc` be sufficient, since we are also using it already in Discover:
https://github.com/elastic/kibana/blob/807625c108a6f9f67875cd891023e8e8a4926ee7/src/plugins/discover/common/utils/sorting/get_sort_for_search_source.ts#L29

Reply:
It would work, it might just lead to duplicate rows sometimes when shard merges occur. For the Logs UI that's a trade-off that we accepted.

Reply:
I think this trade-off is fine